### DISCUSSION TITLE
Part-DB 1.11.0 - LCSC info provider, improved search and API

### DISCUSSION THREAD

**Discussion by jbtronics** [Part-DB/Part-DB-server#537] [Announcements]
# Part-DB 1.11.0

**If you like Part-DB, please consider donating to help further development. See the sponsor button on the [homepage](https://github.com/Part-DB/Part-DB-server).**

## LCSC info provider

Part-DB now can use import part information from [LCSC distributor](https://www.lcsc.com/). See [documentation](https://docs.part-db.de/usage/information_provider_system.html#lcsc) on how to use it.

## Improved search bar

The search bar now makes suggestions for parts while typing. This should make searching much quicker in many cases.

## API improvements
It is now possible to properly create new attachments and parameters on elements using the API like you would expect.

Also, it is now possible to upload files to attachments via API and let Part-DB download URLs. See API docs for more info.

# Full changelog

## New features
* Added LCSC info provider (thanks to @meganukebmp and @frank-f) in PR #526 and #530
* The search bar now shows Part suggestions while typing, allowing for a much quicker part selection in many cases
* Added a search form to the homepage
* Added a button to upload multiple files for attachments at once (see issue #481)
* Added a column to part table to show in which projects a part is used (issue #516)
* Added various project related filters to part tables
* Attachments and parameters can now be properly created via API (see issue #502)
* Allow to upload files and let Part-DB download urls for attachments via API

## Improvements
* Optimized certain SQL queries. Part tables should now load faster in many cases.
* Upgraded hotwire/turbo. The new instaclick feature, already preloads pages on link hover. Link clicks appear much quicker now
* Improved Russian and other translations
* Fixed documentation typos (thanks @codeman256)
* Selection checkboxes in part tables are now real checkboxes which allow for better styling and easier keyboard navigation

## Bugfixes
* Fix problems, if info providers returned attachment urls containing invalid characters (issue #521)

## Miscellaneous
* Upgraded [GNU Unifont](https://unifoundry.com/unifont/index.html) to latest version
* Updated dependencies

## New Contributors
* @meganukebmp made their first contribution in https://github.com/Part-DB/Part-DB-server/pull/526
* @codeman256 made their first contribution in https://github.com/Part-DB/Part-DB-server/pull/528
* @frank-f made their first contribution in https://github.com/Part-DB/Part-DB-server/pull/530

**Full Changelog**: https://github.com/Part-DB/Part-DB-server/compare/v1.10.7...v1.11.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Part-DB/Part-DB-server/releases/tag/v1.11.0'>Part-DB 1.11.0</a>.</em>

**Comment by influx3k** [comment]
There's no way to close the new parts suggestions window that opens when searching for a part.  Even reloading the page or navigating to another page does not get rid of it.

**Comment by tadejg** [comment]
I enabled this info provider last night and got instantly banned - can't even open LCSC in my browser anymore, hours after disabling the provider - responds with 403 Forbidden. Anyone else experiencing any similar issues? I emailed their support to see what they say. It may be a good idea to mention the possibility of a ban in the documentation.

**Reply by jbtronics** [reply]
Maybe you should try a different browser (or incognito tab). In my normal chrome I get an error message from LCSC, in an incognito tab everything works fine.

LCSC seems to behave a bit weird.

**Reply by tadejg** [reply]
Yeah, I literally just thought of that 🙂 turns out that fixes the issue. So it's not the IP being banned, it has something to do with the state of the browser. Looking at the cookies, I found one that seems out of place. It's called "action_track" and contains JSON with a reference to my Part-DB instance.
```
{"from":"http://127.0.0.1:8585/","to":"/"}
```
Deleting this cookie fixes the issue. I wonder why and how that cookie gets created. It's not present, if you use LCSC without Part-DB. If you say your default browser also has similar issues, maybe this is something to look into.

**Comment by frank-f** [comment]
I don't have that problem. I have actually used the provider to grab dozens of parts just yesterday and it's worked every single time.

**Reply by tadejg** [reply]
Interesting. I guess I'll wait for a reply from their support, see what they say about the ban. It just feels like too much of a coincidence to happen right after enabling the info provider.
